libcxx/include/string

```diff
@@ -659,6 +659,13 @@ _LIBCPP_PUSH_MACROS
 #else
 #  define _LIBCPP_STRING_INTERNAL_MEMORY_ACCESS
 #endif
+#ifndef _IF_ASAN
+#  ifndef _LIBCPP_HAS_NO_ASAN
+#    define _IF_ASAN() if _LIBCPP_CONSTEXPR (true)
```

Comment:
Please put this in __config. And you were right, we should name this `_LIBCPP_IF_ASAN()`.

Reply:
Moved back to string file based on: https://github.com/llvm/llvm-project/pull/76165#discussion_r1434348292